tracer/src/Datadog.Tracer.Native/cor_profiler.cpp

```diff
@@ -4333,7 +4333,9 @@ HRESULT STDMETHODCALLTYPE CorProfiler::JITCachedFunctionSearchStarted(FunctionID
         return S_OK;
     }
 
-    *pbUseCachedFunction = true;
+    // Check if this method has been rejitted, if that's the case we don't accept the image
+    bool hasBeenRejitted = this->rejit_handler->HasBeenRejitted(module_id, function_token);
+    *pbUseCachedFunction = !hasBeenRejitted;
```

Comment:
I think it's worth adding a debug log when we hit the "hasBeenRejitted" case

Reply:
done in [46c0a5c](https://github.com/DataDog/dd-trace-dotnet/pull/6184/commits/46c0a5cdb329c0293015f7926407f71df4446fca)

now logs looks like:
```
...
10/23/24 03:16:14.336 PM [5417|3571789] [info] Request ReJIT done for 1 methods
10/23/24 03:16:14.336 PM [5417|3571845] [debug] NGEN Image: Rejected (because rejitted) for Module: Datadog.Trace.Manual, Method:Datadog.Trace.Tracer.get_Instance()
10/23/24 03:16:14.336 PM [5417|3571845] [debug] GetReJITParameters: [moduleId: 12982595232, methodId: 0x06000033]
...
```